src/BIMDataComponents/BIMDataSelect/BIMDataSelectSingle.vue

```diff
@@ -95,17 +109,35 @@ export default {
       type: Boolean,
       default: false,
     },
+    search: {
+      type: Boolean,
+      default: false,
+    },
+    searchPlaceholder: {
+      type: String,
+      default: "Search",
+    },
   },
   emits: ["update:modelValue"],
   data() {
     return {
       isOpen: false,
+      searchText: "",
     };
   },
   computed: {
     displayedValue() {
       return this.optionLabel(this.modelValue);
     },
+    filteredOptions() {
+      if (this.searchText === "") {
+        return this.options;
+      } else {
+        return this.options.filter(option => {
+          return option.toLowerCase().includes(this.searchText.toLowerCase());
```

Comment:
`this.searchText.toLowerCase()` is called for each option.
Can you compute it once before to improve performance?